sqlalchemy_utils/types/choice.py

```diff
@@ -146,9 +146,16 @@ def __init__(self, choices, impl=None):
         if (
             Enum is not None and
             isinstance(choices, type) and
+            issubclass(choices, Enum) and
+            isinstance(impl, types.Integer)
```

Comment:
Well `types.Integer` is a base class for all Integer types, so it should works well.